### ISSUE TITLE
Variables used in the compact function are not reconginized.

### ISSUE THREAD

**Issue by valeryan** [squizlabs/PHP_CodeSniffer#349]
Not sure if this should throw an error or not but her is an example that will not validate using psr2:

``` php
public function verifyEmailUpdate($group, $email, $vcode)
{
    return $this->add_data(compact('email', 'vcode'))
                 ->do_post($this->requestURI($group) . 'updateemail');
}
```

Example Error: Avoid unused parameters such as '$email'.,
